### Create a doc file using python

! pip install python-docx

In [1]:
import  pandas     as     pd
import  datetime
from    datetime   import datetime
import  docx, os

file1  =  'D:/CSI/Members/wrong-domain-csi-member-details2022_05_07_17_13_49.csv'

df1    =   pd.read_csv(file1)
print(df1.shape)
print(df1.head().T)

(133, 7)
                                                                0  \
MemberCode                                               I0001992   
FullName                              Venkataraman R Venkataraman   
emailAddress                             csi00001992@cw-india.org   
ContactDetails                                                0.0   
Address         Plot 4  Street 3 South Sector New Colony ,adam...   
CityName                                              Kanchipuram   
PinCode                                                    600088   

                                            1  \
MemberCode                           I0002142   
FullName          Ramachandran V Ramachandran   
emailAddress         csi00002142@cw-india.org   
ContactDetails                            0.0   
Address         19, Venkatesan Street T Nagar   
CityName                              Chennai   
PinCode                                600017   

                                             

In [2]:
df1.columns

Index(['MemberCode', 'FullName', 'emailAddress', 'ContactDetails', 'Address',
       'CityName', 'PinCode'],
      dtype='object')

In [3]:
Col1_fixed =  ['CSI Member Id', 'Name', 'Email Id', 'Phone', 'Address', 'City', 'Pincode']

In [4]:
Colnamesu  =  ['MemberCode', 'FullName', 'emailAddress', 'ContactDetails', 'Address','CityName', 'PinCode']

In [5]:
df2        =  df1.loc[:, Colnamesu]
df2.shape

(133, 7)

In [6]:
data_df   =  pd.DataFrame()

for j in range(df2.shape[0]):
        df2_df = pd.DataFrame({'SNo' : j, 'CSI Member Id': df2.loc[j, 'MemberCode'],\
                     'Name' : df2.loc[j, 'FullName'],\
                     'Email Id' : df2.loc[j, 'emailAddress'],\
                     'Phone' : df2.loc[j, 'ContactDetails'],\
                     'Address' : df2.loc[j, 'Address'],\
                     'City' : df2.loc[j, 'CityName'],\
                     'Pincode' : df2.loc[j, 'PinCode']} , index = [0])
        data_df   =    data_df.append(df2_df)
                          
###
data_df  =  data_df.set_index('SNo')
print(data_df.shape)

(133, 7)


In [7]:
size  = data_df.shape[0]

In [8]:
data_df

CSI Member Id                               Name  \
SNo                                                    
0        I0001992        Venkataraman R Venkataraman   
1        I0002142        Ramachandran V Ramachandran   
2        I0005576  Balasubramanian K Balasubramanian   
3        I0007199          Venugopalan G Venugopalan   
4        I0014058  Balasubramanian N Balasubramanian   
..            ...                                ...   
128      I0019275  Ramasubramanian V Ramasubramanian   
129      I0004345            Giridharan R Giridharan   
130      I0003081              Devarajan S Devarajan   
131      I0006659                            Rao A P   
132      I0060677                 K Karukkuvel Rajan   

                     Email Id  Phone  \
SNo                                    
0    csi00001992@cw-india.org    0.0   
1    csi00002142@cw-india.org    0.0   
2    csi00005576@cw-india.org    0.0   
3    csi00007199@cw-india.org    0.0   
4    csi00014058@cw-india.org    0.0   
..                        ...    ...   
128  csi00019275@cw-india.org    0.0   
129  csi00004345@cw-india.org    0.0   
130  csi00003081@cw-india.org    0.0   
131  csi00006659@cw-india.org    0.0   
132  csi00060677@cw-india.org    0.0   

                                               Address         City  Pincode  
SNo                                                                           
0    Plot 4  Street 3 South Sector New Colony ,adam...  Kanchipuram   600088  
1                        19, Venkatesan Street T Nagar      Chennai   600017  
2    Director Akshay Business Services P Ltd 33 Fir...      Chennai   600020  
3                      1a Mahalakshmi St East Tambaram  Kanchipuram   600059  
4     23/6 Kamal Apartments Ist Cross Street R A Puram      Chennai   600028  
..                                                 ...          ...      ...  
128      Flat 38 No 4 4th Main Road Gandhi Nagar Adyar      Chennai   600020  
129  Block No.8 , Flat No.9 , Second Floor,brindava...   Tiruvallur   600062  
130            3 Park Avenue K P Puram Green Ways Road      Chennai   600028  
131  S2, ramakrishna 47 Kandasamy Street, Pallipatt...      Chennai   600113  
132     4/49, Indira Street Varadarajpuram,ponnamalari  Kanchipuram   600056  

[133 rows x 7 columns]

In [9]:
data_df.columns

Index(['CSI Member Id', 'Name', 'Email Id', 'Phone', 'Address', 'City',
       'Pincode'],
      dtype='object')

In [10]:
test_list =  [(('CSI Member Id', data_df.loc[0, 'CSI Member Id']),\
              ('Name', data_df.loc[0, 'Name']),\
               ('Email Id', data_df.loc[0, 'Email Id']),\
               ('Phone', data_df.loc[0, 'Phone']),\
               ('Address', data_df.loc[0, 'Address']),\
               ('City', data_df.loc[0, 'City']),\
               ('Pincode', data_df.loc[0, 'Pincode']))]
test_list

[(('CSI Member Id', 'I0001992'),
  ('Name', 'Venkataraman R Venkataraman'),
  ('Email Id', 'csi00001992@cw-india.org'),
  ('Phone', 0.0),
  ('Address', 'Plot 4  Street 3 South Sector New Colony ,adambakkam'),
  ('City', 'Kanchipuram'),
  ('Pincode', 600088))]

In [11]:
os.chdir('D:/CSI/Members/Output')

###
# Table data in a form of list
###
dataList = [[('MemberCode', 'I0001992'),\
        ('FullName', 'Venkataraman R'),\
        ('emailAddress', ' '),\
        ('Phone Number', '0'),\
        ('Address', 'Plot 4  Street 3 South Sector New Colony ,Adambakkam'),\
        ('CityName', 'Kanchipuram'),\
        ('PinCode', '600088')],
       [('MemberCode', 'I0002142'),\
        ('FullName', 'Ramachandran V'),\
        ('emailAddress', ' '),\
        ('Phone Number', '0'),\
        ('Address', '19, Venkatesan Street T Nagar'),\
        ('CityName', 'Chennai'),\
        ('PinCode', '600017')]
       ]

In [12]:
dtd  =  'Date: ' + str(datetime.now().strftime('%d-%m-%Y'))
print(dtd)

Date: 10-05-2022


In [13]:
###
# Create an instance of a word document
###
doc  =  docx.Document()

In [14]:
for l  in  range(size):
    data =  (('CSI Member Id', data_df.loc[l, 'CSI Member Id']),\
              ('Name', data_df.loc[l, 'Name']),\
               ('Email Id', data_df.loc[l, 'Email Id']),\
               ('Phone', data_df.loc[l, 'Phone']),\
               ('Address', data_df.loc[l, 'Address']),\
               ('District / City', data_df.loc[l, 'City']),\
               ('Pincode', data_df.loc[l, 'Pincode']))
    ###
    # Adding heading in the first row of the table 
    ###
    doc.add_heading('Confirmation of your contact details', 0) 
    ### 
    # Adding a paragraph
    ###        
    doc.add_paragraph(dtd) 
    doc.add_heading('\nDear CSI Member', 3)
    doc.add_paragraph('\nGreetings from CSI Chennai Chapter!')    
    doc.add_paragraph('Your contact details as per our records are given below:')

    # Creating a table object
    ###
    table  = doc.add_table(rows = 1, cols = 2)
    # Add a Title to the document
    row               =  table.rows[0].cells
    row[0].text       = 'Column'
    row[1].text       = 'Details'
    ###
    for idx, txt in data:
        ###
        # Adding a row and then adding data in it
        ###
        row          =  table.add_row().cells
        row[0].text  =  idx
        row[1].text  =  str(txt)
        ###
        
    doc.add_paragraph('\n\nPlease confirm if they are correct. If they are incorrect or changed,please send email us at info@csichennai.org with your Membership Id and the correct details.\n\n')
    doc.add_paragraph('\n\nWarm regards')  
    doc.add_paragraph('TV Kamalak Kannan, Chair, CSI Chennai Chapter\n')
    doc.add_paragraph('\n\nThis is a computer generated document requiring no signature')
    # Adding style to a table
    ###
    table.style   =  'Light Grid'
    # Adding a page break
    doc.add_page_break()

In [15]:
###
# Now save the document to a location
###
output_file_name =  'undelivered_csi-member-details' + str(datetime.now().strftime('%Y_%m_%d_%H_%M_%S')) + '.docx'
doc.save(output_file_name)
###
# END
###